modules/collect/src/main/java/com/opengamma/collect/id/LinkResolutionException.java

```diff
@@ -18,21 +18,10 @@
   /**
    * Creates the exception passing the failed link.
    *
-   * @param link  the link which could not be resolved
-   * @param causeMessage  the reason why the link could not be resolved
+   * @param message  the reason why the link could not be resolved, null tolerant
    */
-  public LinkResolutionException(Link<?> link, String causeMessage) {
-    super("Unable to resolve link: " + link + " - " + String.valueOf(causeMessage));
-  }
-
-  /**
-   * Creates the exception passing the failed link.
-   *
-   * @param link  the link which could not be resolved
-   * @param cause  the reason why the link could not be resolved
-   */
-  public LinkResolutionException(Link<?> link, Failure cause) {
-    super("Unable to resolve link: " + link + " - " + (cause != null ? cause.getMessage() : "<no cause>"));
+  public LinkResolutionException(@Nullable String message) {
```

Comment:
The constructor is used from one place, and moving the exception message there simplifies things. If the exception stored the link internally, a case might be made, but that would have limited value, so wasn't worth doing.